src/components/SelectionList/BaseSelectionList.js

```diff
@@ -0,0 +1,403 @@
+                        {shouldShowTextInput && (
+                            <View style={[styles.ph5, styles.pb3]}>
+                                <TextInput
+                                    ref={textInputRef}
+                                    label={textInputLabel}
+                                    accessibilityLabel={textInputLabel}
+                                    accessibilityRole={CONST.ACCESSIBILITY_ROLE.TEXT}
+                                    value={textInputValue}
+                                    placeholder={textInputPlaceholder}
+                                    maxLength={textInputMaxLength}
+                                    onChangeText={onChangeText}
+                                    keyboardType={keyboardType}
+                                    selectTextOnFocus
+                                    spellCheck={false}
+                                    onSubmitEditing={selectFocusedOption}
+                                />
+                            </View>
+                        )}
+                        {Boolean(headerMessage) && (
+                            <View style={[styles.ph5, styles.pb5]}>
+                                <Text style={[styles.textLabel, styles.colorMuted]}>{headerMessage}</Text>
+                            </View>
+                        )}
+                        {flattenedSections.allOptions.length === 0 && showLoadingPlaceholder ? (
+                            <OptionsListSkeletonView shouldAnimate />
+                        ) : (
+                            <>
+                                {!headerMessage && canSelectMultiple && shouldShowSelectAll && (
+                                    <PressableWithFeedback
+                                        style={[styles.peopleRow, styles.userSelectNone, styles.ph5, styles.pb3]}
+                                        onPress={onSelectAll}
+                                        accessibilityLabel={translate('workspace.people.selectAll')}
+                                        accessibilityRole="button"
+                                        accessibilityState={{checked: flattenedSections.allSelected}}
+                                    >
+                                        <Checkbox
+                                            accessibilityLabel={translate('workspace.people.selectAll')}
+                                            isChecked={flattenedSections.allSelected}
+                                            onPress={onSelectAll}
+                                        />
+                                        <View style={[styles.flex1]}>
+                                            <Text style={[styles.textStrong, styles.ph5]}>{translate('workspace.people.selectAll')}</Text>
+                                        </View>
+                                    </PressableWithFeedback>
+                                )}
+                                <SectionList
```

Comment:
Coming from https://github.com/Expensify/App/issues/26243:
This refactor caused minor inconsistency with OptionsList.
`stickySectionHeadersEnabled` prop is enabled by default in SectionList.
`OptionsList` disabled it but `SelectionList` from this PR didn't.